server/users.ts

```diff
@@ -1261,13 +1261,20 @@ export class User extends Chat.MessageContext {
 				return false;
 			}
 		}
-		if ((room as GameRoom).tour) {
+		if (room.tour) {
 			const errorMessage = (room as GameRoom).tour!.onBattleJoin(room as GameRoom, this);
 			if (errorMessage) {
 				connection.sendTo(roomid, `|noinit|joinfailed|${errorMessage}`);
 				return false;
 			}
 		}
+		if (roomid.startsWith('battlechat-')) {
+			const battleRoom = Rooms.get(roomid.replace('battlechat-', 'battle-'));
+			if (battleRoom?.battle?.active && this.id in battleRoom.battle.playerTable) {
+				connection.sendTo(roomid, `|noinit|joinfailed|You can't join the battle chat for a battle you're currently playing.`);
+				return false;
+			}
+		}
```

Comment:
It would be easy to circumvent this simply by making an alt; I don't know how serious we are about not letting the players see the chat, but it might be worth considering using _some_ punishments tools (checking IPs and past names, at least) to make evasion less trivial.